### ISSUE TITLE
Query regarding layer-wise analog conversion

### ISSUE THREAD

**Issue by IannoIITR** [IBM/aihwkit#529]
I had a model with grouped convolutions(which is not supported in aihwkit). So I tried converting individual layers from the model dict to analog. 
for eg.
model.backbone.input_stem.op_list[0] = convert_to_analog(model.backbone.input_stem.op_list[0],rpu_config)
model.backbone.input_stem.op_list[1].main.point_conv = convert_to_analog(model.backbone.input_stem.op_list[1].main.point_conv ,rpu_config)
model.backbone.stages[0].op_list[0].main.inverted_conv = convert_to_analog(model.backbone.stages[0].op_list[0].main.inverted_conv ,rpu_config)
model.backbone.stages[0].op_list[0].main.point_conv = convert_to_analog(model.backbone.stages[0].op_list[0].main.point_conv ,rpu_config)

so basically replacing the existing floating point layer in the model with an equivalent one converted to analog. 

I wanted to know if this is a valid approach or is there any problem in it as it is working for some models but in others there is a  large drop in accuracy. (single layer drops accuracy of pre-trained model to 0, even when rpu config is near ideal [32 bit resolution and no noise values])

**Comment by maljoras** [comment]
Hi @IannoIITR ,
right grouped convolution is currently not supported. Typically you want to just use `convert_to_analog(model, rpu_config)` for the full model, it will look for the `Conv2D` layers to be replaced. 

In your approach, is the e.g. `point_conv` still a grouped convolution? Then it will not work. The problem with grouped convolution is that they map poorly to crossbar tiles, as they result in block-diagonal kernel matrices. One could put the blocks onto individual tiles, but that would be come quickly inefficient. Or one could use a full tile for the non-grouped conv kernel matrix and then  "program" approx zeros in the off-diagonal values. For inference it would be fast, but for training it would need some sequential masking of input and gradients for each group. 

So while problematic for large number of groups, for small number of groups it could be reasonable. Handling conv groups could be a good feature to add to the toolkit in future.
